rpc/wrtc_base_channel.go

```diff
@@ -131,13 +131,14 @@ func newBaseChannel(
 				connIDMu.Lock()
 				connID = connInfo.ID
 				connIDMu.Unlock()
-				logger.Debugw("connection state changed",
+				logger.Infow("Connection state changed",
```

Comment:
I've bumped this log and the selected candidate pair to INFO. These will occur for both answerers and clients for connection states of `new`, `checking`, `connected` and `completed`; see the PR description for examples of what these look like.

Reply:
I buy this gets us better info when we connect. Are we guaranteed to get the same information (or more) when we fail to connect?

I don't have a strong model for understanding when/why connection states change and how that relates to (complete) ice candidate information.

Reply:
> Are we guaranteed to get the same information (or more) when we fail to connect?

...Nope! 😆 Great call out.

> I don't have a strong model for understanding when/why connection states change and how that relates to (complete) ice candidate information.

With what I have here, we'll only list local and remote ICE candidates when the ICE connection state is "new", "checking", "connected" or "completed". A failure to connect may represent a halting of the connection establishment process in some interstitial state, particularly between "new" (or "checking") and "connected". So, with the logs in this PR, we may have stale sense of ICE candidates upon connection failure. I'll think about how to fix that.

Reply:
So this patch seems to be changing a few things:
- It's making the output more structured
- It's using more of the information available from the webrtc candidate stats
  - Removing the "live" capturing from we did with answerer stats
- It's changing how/when we log this information to when connection state changes

I'm not sure if I understand the motivations for the last. If adjusting that bit becomes complex, let's try to get the first two in. Then we can revisit what to do about the third when I put up a PR that mucks with error handling/determining whether a connection succeeds.

Reply:
> It's changing how/when we log this information to when connection state changes

I've added in logs for when connection establishment fails. I would argue that with those logs, we're logging the state of ICE at all important inflection points. Happy to discuss more offline or continue to discuss in your eventual PR that changes the definition of "success".

Reply:
That makes sense. I'm partially worried this will be a bit too many "small" log lines. I'm bought into these being the important events we should be latching onto.

I think what you have for now is fine. And if it looks chatty -- we can move to more of a "accumulate everything -- log once" direction.